packages/mui-base/src/Slider/SliderOutput/useSliderOutput.ts

```diff
@@ -0,0 +1,43 @@
+'use client';
+import * as React from 'react';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import { useSliderContext } from '../Root/SliderProvider';
+import { UseSliderOutputParameters, UseSliderOutputReturnValue } from './SliderOutput.types';
+/**
+ *
+ * API:
+ *
+ * - [useSliderOutput API](https://mui.com/base-ui/api/use-slider-output/)
+ */
+function useSliderOutput(parameters: UseSliderOutputParameters): UseSliderOutputReturnValue {
+  const { 'aria-live': ariaLive = 'off', rootRef } = parameters;
+
+  const { subitems } = useSliderContext();
```

Comment:
Hooks should avoid using contexts since they're the responsibility of component authors. Instead they should receive everything as explicit params.

Reply:
Like this? https://github.com/mui/base-ui/pull/373/commits/dfd9d44fd731d410fdd9a79fd81fb53afdd9a459

I feel like this makes the hooks harder to use for hook users e.g. it would expose some internals (e.g. a internal [`changeValue`](https://github.com/mui/base-ui/blob/dfd9d44fd731d410fdd9a79fd81fb53afdd9a459/packages/mui-base/src/Slider/Root/useSliderRoot.ts#L214) function, various `setState`s) that users didn't need to care about at all

Maybe we could use context in the hooks for things like these?
But for things like `min`, `max`, `orientation` etc that are also component props, make them explicit params of the hook

What do you think @atomiks @michaldudak ?

BTW I saw the various Tab hooks also use the context, e.g. [useTabsList](https://github.com/mui/base-ui/blob/master/packages/mui-base/src/Tabs/TabsList/useTabsList.ts#L32-L39)

Reply:
We could export contexts, which would require hook users to set them up, though it may restrict the component APIs they want to build. Hooks are mostly considered an escape hatch to begin with, so I didn't think passing all those props was too bad.

Reply:
👌 Updated